### PR TITLE
Add c++20 character types

### PR DESCRIPTION
**Pull request by i-garrison**
This change adds `char8_t` character type from c++20

### CODE REVIEW

**Comment by jonahgraham** [comment]
The build on this is failing, best guess is some inlined constant or change of used internals from cdt core has changed causing the failure: 

```
Error:  Failed to execute goal org.eclipse.tycho:tycho-p2-plugin:3.0.3:p2-metadata (baselinereplace-p2-metadata) on project org.eclipse.cdt.ui: baseline and build artifacts have same version but different contents
Error:     no-classifier: different
Error:        org/eclipse/cdt/internal/ui/refactoring/rename/ASTManager.class: different
Error:  -> [Help 1]
```

It would be nice to know why it has changed, but either way, the version of org.eclipse.cdt.ui needs to be bumped.

**Comment by i-garrison** [comment]
> org.eclipse.cdt.ui needs to be bumped.

Hi @jonahgraham either I bumped the wrong piece in [4411c45](https://github.com/eclipse-cdt/cdt/pull/342/commits/4411c45f6529c3c9135d264eb9cffc17b940562a) or it did not helped here.

**Comment by jonahgraham** [comment]
Weird goingson now. The current code passes code cleanliness without the bump, so not sure what is going on and I won't spend much time on that. I will review this later.

**Comment by i-garrison** [comment]
Well there actually was a bump in Navigator cleanup 47275c38848aca2c27477b9a578564e198abcee4
I just did a rebase onto main and tests are now passing.

**Comment by jonahgraham** [comment]
Thanks for figuring out what happened.

File: core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/dom/parser/cpp/CPPASTLiteralExpression.java [hunk 1]
```diff
@@ -437,7 +444,13 @@ public Kind getBasicCharKind() {
 			return Kind.eChar32;
 		case 'u':
 			// Bug 526724 u8 should result in Kind.eChar
-			if (fLiteral[1] != '8') {
+			if (fLiteral[1] == '8') {
```
Comment (jonahgraham) [inline core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/dom/parser/cpp/CPPASTLiteralExpression.java:447]: Please remove `//$FALL-THROUGH$` comment on line 456

**Review by jonahgraham — COMMENTED** [review]
This LGTM. All the changes made seem good. There is still the possibility that some other areas need modification, but I don't have the knowledge to know what may be missing here.
 
I will make the small change mentioned and submit today for M1.

**Review by jonahgraham — APPROVED** [review]

**Pull request merged**